lib/kamal/cli/server.rb

```diff
@@ -32,6 +32,14 @@ def bootstrap
           if execute(*KAMAL.docker.superuser?, raise_on_non_zero_exit: false)
             info "Missing Docker on #{host}. Installing…"
             execute *KAMAL.docker.install
+            begin
+              execute *KAMAL.docker.add_group
+
+              # If the groups change, the session is terminated to force a re-login.
+              # Catch the resulting IOError and inform the user
+            rescue IOError
```

Comment:
Is there a more specific error we can rescue here?

Reply:
That is the specific error. We get back "IOError: closed stream" from the terminated ssh session. The only subclass of IOError is EOFError. It's not a part of the SystemCallError tree.